### PR TITLE
Task/student2

### PR DESCRIPTION
**Pull request by electrovladyslav**
Deployed version - https://electrovladyslav.github.io/react-mentors-test-2024/
Task description - https://github.com/vladyslav-yermolin-moc/react-mentors-test-2024/issues/3#issue-2190644484

### CODE REVIEW

File: src/ProductPage.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    } else if (sortBy === 'priceHighToLow') {
+      sortedProducts.sort((a, b) => b.price - a.price);
+    }
+    setProducts(sortedProducts);
+  };
+
+  return (
+    <div>
+      <input type="text" value={searchTerm} onChange={handleSearchChange} placeholder="Search..." />
+      <select value={sortBy} onChange={handleSortChange}>
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:80]: create shared custom input and reuse

File: src/ProductPage.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        <option value="">Sort By</option>
+        <option value="priceLowToHigh">Price: Low to High</option>
+        <option value="priceHighToLow">Price: High to Low</option>
+      </select>
+      <label>
+        New
+        <input type="checkbox" name="new" checked={filterOptions.new} onChange={handleFilterChange} />
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:87]: also here create shared component and not checkbox 1 by 1 
your component get config file with list of checkboxs and effect data object whe do some event

File: src/ProductPage.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import css from './ProductPage.module.css';
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:2]: better name producPageStyles

File: src/ProductPage.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+interface Product {
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:4]: create new file and its type not interface

File: src/ProductPage.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    new: false,
+    old: false
+  });
+
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  const fetchProducts = async () => {
+    // Симуляція отримання даних з сервера
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:33]: comment not UKR only ENG

File: src/ProductPage.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    const response = await fetch('https://api.escuelajs.co/api/v1/products?limit=0&offset=0');
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:34]: `https://api.escuelajs.co/api/v1/products?limit=${limit}&offset=${offset}`

need custom url

File: src/ProductPage.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  const handleFilterChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, checked } = event.target;
+    setFilterOptions({ ...filterOptions, [name]: checked });
+    filterProducts();
+  };
+
+  const filterProducts = () => {
+    let filteredProducts = [...products];
+    if (filterOptions.new) {
+      filteredProducts = filteredProducts.filter((product) => product.creationAt > '2021-01-01');
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:59]: into const config 2021-01-01

File: src/ProductPage.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    if (filterOptions.old) {
+      filteredProducts = filteredProducts.filter((product) => product.creationAt < '2021-01-01');
+    }
+    setProducts(filteredProducts);
+  };
+
+  const sortProducts = () => {
+    const sortedProducts = [...products];
+    if (sortBy === 'priceLowToHigh') {
```
Comment (vitalii-mel-work) [inline src/ProductPage.tsx:69]: config all types

**Comment by vitalii-mel-work** [comment]
good job , but need review one more time after your updates

**Comment by vladyslav-yermolin-moc** [comment]
Hey, @vitalii-mel-work, thanks for the quick code review. We will discuss it in our sync. Have a nice day!

**Pull request closed without merging**